src/Evolution/DiscontinuousGalerkin/Actions/ApplyBoundaryCorrections.hpp

```diff
@@ -314,14 +354,40 @@ void ApplyBoundaryCorrections<Metavariables>::apply_global_time_stepping(
                 direction.side() == Side::Upper ? matrices.second
                                                 : matrices.first;
             apply_matrices(make_not_null(&get(face_det_jacobian)),
-                           interpolation_matrices, volume_det_jacobian,
+                           interpolation_matrices, get(volume_det_jacobian),
                            volume_mesh.extents());
 
             evolution::dg::lift_boundary_terms_gauss_points(
                 dt_variables_ptr, volume_det_inv_jacobian, volume_mesh,
                 direction, dt_boundary_correction, magnitude_of_face_normal,
                 face_det_jacobian);
           }
+
+          ASSERT(not local_time_stepping,
+                 "Must return lifted data when using local time stepping");
+          return {};
+        };
+
+        for (auto& mortar_id_and_data : *mortar_data_ptr) {
+          // Cannot use structured bindings because of a compiler bug where
+          // they cannot be captured by lambdas.
+          const auto& mortar_id = mortar_id_and_data.first;
+          const auto& mortar_data = mortar_id_and_data.second;
+          const auto& direction = mortar_id.first;
+          if (UNLIKELY(mortar_id.second ==
+                       ElementId<volume_dim>::external_boundary_id())) {
+            ERROR(
+                "Cannot impose boundary conditions on external boundary in "
+                "direction "
+                << direction
+                << " in the ApplyBoundaryCorrections action. Boundary "
+                   "conditions are applied in the ComputeTimeDerivative "
+                   "action "
+                   "instead. You may have unintentionally added external "
+                   "mortars in one of the initialization actions.");
+          }
+          mortar_id_ptr = &mortar_id;
+          compute_correction_coupling(mortar_data, mortar_data);
```

Comment:
For my understanding: why does this lambda take the same arg twice?

Reply:
with LTS, the local and neighbor mortar data are separate objects, but the actual boundary correction computation is the same with LTS and GTS. In order to share that code, `compute_correction_coupling` needs to take two mortar data, the first is used for the local data, the second for the neighbor data. With GTS the local and neighbor data are stored in the same object, so I pass the same object to both arguments

Reply:
Thanks for clarifying!